doc/src/tutorial/intro.rst

```diff
@@ -219,3 +219,85 @@ if you wish to automate or extend them, it is difficult to do.  With SymPy,
 you can just as easily use it in an interactive Python environment or import
 it in your own Python application.  SymPy also provides APIs to make it easy
 to extend it with your own custom functions.
+
+
+Understanding SymPy Ondřej's approach
+=====================================
```

Comment:
I would also delete this. It references outdated stuff (that flatten routine doesn't look like that at all anymore).  Did you, as a new user to SymPy, find this section useful?